packages/hooks/src/useAnalytics.ts

```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import { Analytics, TEvents } from '@deriv/analytics';
+import { useStore } from '@deriv/stores';
+import { removeNullOrUndefinedValues } from '@deriv/shared';
+
+export type TTrackRealAccountSignup = Omit<
+    TEvents['ce_real_account_signup_form'],
+    'landing_company' | 'user_choice'
+> & {
+    user_choice?: string | object;
+};
+
+const useAnalytics = () => {
+    const { ui, client } = useStore();
+    const { is_mobile, real_account_signup_target } = ui;
+
+    React.useEffect(() => {
+        client?.loginid &&
+            Analytics.setAttributes({
+                device_type: is_mobile ? 'mobile' : 'desktop',
+                account_type: client?.loginid?.slice(0, 2),
```

Comment:
as far as I know all events already handled by us
let's test this case together with and without this useEffect

Reply:
suggested by @amir-deriv however we can test this and see if its needed or not